dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -746,4 +748,80 @@ private static AcknowledgedResponse setClusterAsReadOnly(final boolean value) {
                         .putSettings(request, RequestOptions.DEFAULT)
         );
     }
+
+    /**
+     * Method to Test: {@link ESContentletAPIImpl#checkin(Contentlet, User, boolean)}
+     * When:
+     * - Create a Content Type with a MANY_TO_MANY relationship to itself
+     * - Create tree contents: A, B, C
+     * - Related content A with B and C
+     * - Related content B with A and C
+     * - Related content C with A and B
+     * Should: All the related content should be saved right
+     */
+    @Test
+    public void selfRelatedContents() throws DotDataException {
+        final Host host = new SiteDataGen().nextPersisted();
+        final ContentType contentType =  new ContentTypeDataGen()
+                .host(host)
+                .nextPersisted();
+
+        final Relationship relationship = new FieldRelationshipDataGen()
+                .child(contentType)
+                .parent(contentType)
+                .nextPersisted();
+
+        Contentlet contentletA = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        Contentlet contentletB = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        Contentlet contentletC = new ContentletDataGen(contentType)
+                .host(host)
+                .nextPersisted();
+
+        contentletA = releteContent(relationship, contentletA, contentletB, contentletC);
+        contentletB = releteContent(relationship, contentletB, contentletA, contentletC);
+        contentletC = releteContent(relationship, contentletC, contentletA, contentletB);
+
+        contentletA.setProperty(relationship.getChildRelationName(), Arrays.asList(contentletB, contentletC));
+        contentletB.setProperty(relationship.getChildRelationName(), Arrays.asList(contentletA, contentletC));
```

Comment:
why a re you setting the relationship for contentletB and C if only A is checked in?

Reply:
done https://github.com/dotCMS/core/pull/20558/commits/fa048b2c56dcf34de41da210dfa35a2e2c0224b5#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9L791